### ISSUE TITLE
Make OSD metric symbols more... metric ;) Add support for fonts stored in PNG with transparency

### ISSUE THREAD

**Issue by robertrypula** [bri3d/msp-osd#90]
For some reason this kills my eyes, few examples:

![image](https://user-images.githubusercontent.com/6407731/207851040-0eed08ae-4ad2-43d8-bc16-cb24a18d7fe3.png)

Maybe there are some places in the world that this is used on daily basis but in general there are some rules in metric system how to express units:

| full name  | correct when symbols are used | incorrect |
| ------------- | ------------- |----|
| meter  | m  | M, m., mtr, mtrs |
| kilometer  | km  | Km, km., KM, kms, K, k |
| milliampere hours  | mAh  | mamph, mah, mAhr |
| second  | s | sec, S, SEC, sec., s., S. |
| hour | h | hr, hrs, HR, h., HR., HRS. |
| Ampere | A | Amp (used in spoken language but technically incorrect when used as unit symbol) |
| Volt | V |  |

Also metric system uses "/" for "per" as this **scales up very well for any unit**. We don't have to mix symbols with "p" character when expressing any combination of units. Also there is no logic behind expressing km/h as kph but using "/" for the other units like m/s, cm/s, mm/s (mps, cmps, mmps is like WAT? :D). Consistency is the key so it would be nice to stick to just one character like "/":

| full name | correct when symbols are used| incorrect usage |
|-----|---------|------------------|
| meters per second | m/s | M/s (M is mega, m is meter), m/sec (in metric system symbol for second is just "s"), M/sec |
| kilometers per hour| km/h | kph, kmh, kmph (kilo mile per hour? :D) |
| centimeters per second | cm/s | cmps (WAT?) |
| degrees per second | deg/s | dps?, degps? oh noo ;) |

Metric symbols are case sensitive. Maybe M/S is kind of okay because we might not have lowercase fonts but mixing it like M/s is not correct. It's not "Mega per second" :P

More info here https://usma.org/correct-usage-of-the-si

This lead me to the question. Is there any way where I can open those bin files in for example Photoshop and edit those graphics? This really kills my eyes and destroys experience :D

**Comment by hickscorp** [comment]
A guess here - but as it is possible to use custom fonts for the OSD by placing them at the root of your SD card, and since these units are actually each just a single glyph in the font file, it is very likely that you should be able to patch your favourite font file and drop it on the SD card.

I might give it a shot this weekend, as this is also something that disturbs me - it doesn't hurt my eyes, but having to rely on my analytic brain while flying kind of breaks my flow :)

**Comment by Knifa** [comment]
lmao ok, if you feel so strongly about it fair play 😄 [here's an update for material-osd](https://github.com/Knifa/material-osd/releases/tag/v0.0.4).

the primary reason for me was because `km/h` is longer than `kph` so it wouldn't fit in the glyph box unless it was scaled down, which means all the other units need scaled down to match. they end up pretty small on HD. i'm sure it's fine though.

![image](https://user-images.githubusercontent.com/622619/213866255-56f7d5f9-aadf-4ff0-a33c-af00a28682f5.png)
![image](https://user-images.githubusercontent.com/622619/213866294-3bb2974f-7228-4364-8b84-8faed3a36113.png)


the fonts aren't owned by this repo fwiw, but a fun post none-the-less.

**Comment by robertrypula** [comment]
Thanks! <3 I'll check it in more details over next weekend!

**Comment by robertrypula** [comment]
> A guess here - but as it is possible to use custom fonts for the OSD by placing them at the root of your SD card, and since these units are actually each just a single glyph in the font file, it is very likely that you should be able to patch your favourite font file and drop it on the SD card.
> 
> I might give it a shot this weekend, as this is also something that disturbs me - it doesn't hurt my eyes, but having to rely on my analytic brain while flying kind of breaks my flow :)

Yeah I wanted to update them in Photoshop but I failed on finding the way on editing the ***.bin** file :) If googles could read **PNG with transparency** that would be awesome :D

**Comment by vicewize** [comment]
Ill try to do my best to update my font, i'm an engineer and only use the metric system and I absolutely hate the imperial system (im from USA) so i get it 100%. I can't stand it when I have to work on a project in Fusion360 in inches or worse fractional inches.... Remnant of British imperialism and Americans are stubborn as f, to change anything, even though most people know its so much better and 10x simpler...  I'll read this thread more throughly when i have time. :-) It was basically a size issue because when the glyths get down past a certain size, its gets really hard to read in the googles no matter how "pixel perfect" I make them in illustrator when I rasterize them, but ill see what I can do. Ive been super busy and haven't made any new fonts or updates but was planning to but you know how it is... life.. Thanks for trying my font btw :-)

**Comment by robertrypula** [comment]
I'm now in the field testing all the fonts (finally it's not raining). I'll make comparison and l make a list :)

**Comment by robertrypula** [comment]
Hello @vicewize 
I finally had some time to continue on this topic. Please find below my suggestions. I might be wrong in my thinking so I'm open for discussion :)

| current behavior  | suggestion | comment |
:-------------------------:|:-------------------------:|:-------------------------:
![image](https://user-images.githubusercontent.com/6407731/218031775-a5f17532-9fe7-49d5-8459-c983e0a2c0fa.png) | ![image](https://user-images.githubusercontent.com/6407731/218030902-ed97952d-6ade-4aab-9641-825a8a886264.png) | SI uses "/" for "per"
![image](https://user-images.githubusercontent.com/6407731/218031452-fc398fdc-eff5-4843-a4f9-f486cef137b8.png) | ![image](https://user-images.githubusercontent.com/6407731/218031633-e12fa33b-ed80-4cc2-8dc6-28a05ba04128.png) | M is mega, SI symbol for second is just "s" not sec
![image](https://user-images.githubusercontent.com/6407731/218031874-05833e9b-2d41-431b-9ba0-2c61c2e30b02.png) | ![image](https://user-images.githubusercontent.com/6407731/218031957-d33e353b-e53c-4b80-9a00-2f6e4b441848.png) | Inconsistency - if we use lower case charracters in other places when km should be with lowercase characters
![image](https://user-images.githubusercontent.com/6407731/218032338-bf70203e-f639-4211-a346-d4e78779fef9.png) | ![image](https://user-images.githubusercontent.com/6407731/218032532-a2329218-5766-432c-a4fe-83cb46110166.png) | "ft" for case consistensy and "s" for second. Since this is imperial unit I guess fps is also correct (according to Wikipedia)
![image](https://user-images.githubusercontent.com/6407731/218032794-07460049-958b-47bc-a6e8-7ee8ab9aee2f.png) | ![image](https://user-images.githubusercontent.com/6407731/218032944-78992187-bb41-4683-a0ce-cd4f17dd817f.png) | Units should be upper case
![image](https://user-images.githubusercontent.com/6407731/218033114-85bb256c-c157-4d1e-b940-d4dd9a70567b.png) | ![image](https://user-images.githubusercontent.com/6407731/218033179-72589eee-1f40-4c4b-ac3b-f40bd5409b0d.png) | Dot looks like minus, IMHO it looks weird on units especially here: ![image](https://user-images.githubusercontent.com/6407731/218033512-59630bfd-51d4-42a1-be6c-653179fc5dba.png) It's like - 0 _ 3 m
![image](https://user-images.githubusercontent.com/6407731/218033865-a5b5505b-a2b6-44e9-baac-10d72f95744a.png) | ![image](https://user-images.githubusercontent.com/6407731/218033902-0979cfa1-1349-41d0-a662-f23a6e3e379c.png) | SI symbol for Ampere is uppercase "A". General Rule in SI is that we use upper case for units when name comes from a real person (like "W" for Mr. Watt, "V" for Mr. Volt, "A" for Mr. Ampere)
![image](https://user-images.githubusercontent.com/6407731/218034126-3dc5ca81-4f3d-4572-bb62-16f97e0e95ea.png) | ![image](https://user-images.githubusercontent.com/6407731/218034537-c4026021-9e6a-4704-8da3-3cdf151443a0.png) | Here I'm not sure, upper case it's easier to read but we have inconsistency with places where lower/upper case are used
![image](https://user-images.githubusercontent.com/6407731/218034300-78f63d5b-ff88-49eb-944f-975c826464a0.png) | ![image](https://user-images.githubusercontent.com/6407731/218034442-f1630dec-f654-4fce-962a-dc0e02198d25.png) | Same as above
![image](https://user-images.githubusercontent.com/6407731/218035188-7cf48823-7602-4060-8e6c-a47cafddd730.png) | ![image](https://user-images.githubusercontent.com/6407731/218035258-d4900cf2-9999-4dfe-ba03-96c0fe1f7e8d.png) | IMHO the / is not needed as it looks now as "ON per minute" and "FLY per min" which doesn't make sense :D 

In general I found this website very helpful https://usma.org/correct-usage-of-the-si. Also Wikipedia is cool to double check the units.

I was also testing all the other fonts. It would be super cool to add CHROMA version of the VICEWIZE Italic font. You could use existing tools to overlay it over GoPro footage. Check my video at the end:

https://youtu.be/C83AlfBp-9k?t=186

[![YouTube](https://user-images.githubusercontent.com/6407731/218035706-bb116360-a78c-46c5-8ad1-3568f15fb2fb.png)](https://youtu.be/C83AlfBp-9k?t=186)
